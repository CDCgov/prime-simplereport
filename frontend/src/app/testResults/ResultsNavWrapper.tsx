import React, { ReactElement } from "react";

import ResultsNav from "./ResultsNav";

type Props = {
  children: ReactElement;
};

const ResultsNavWrapper = ({ children }: Props) => {
  return (
    <div className="prime-home flex-1">
      <div className="grid-container results-wide-container">
        <ResultsNav />
        {children}
      </div>
    </div>
  );
};

export default ResultsNavWrapper;
