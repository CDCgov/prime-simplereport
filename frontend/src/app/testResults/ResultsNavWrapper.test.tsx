import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import ResultsNavWrapper from "./ResultsNavWrapper";

jest.mock("uuid", () => ({
  v4: jest.fn(() => "static-uuid"),
}));
describe("ResultsNav", () => {
  it("displays the results sub nav when user has the permission", () => {
    const { container } = render(
      <MemoryRouter>
        <ResultsNavWrapper>
          <>Testing 123</>
        </ResultsNavWrapper>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("doesnt display the results sub nav when user doesnt have the permission", () => {
    const { container } = render(
      <MemoryRouter>
        <ResultsNavWrapper>
          <>Testing 123</>
        </ResultsNavWrapper>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});