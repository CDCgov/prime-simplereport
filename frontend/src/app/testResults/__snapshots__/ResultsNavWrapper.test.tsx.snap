// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsNav displays the results sub nav when user has the permission 1`] = `
<div>
  <div
    class="prime-home flex-1"
  >
    <div
      class="grid-container results-wide-container"
    >
      <nav
        aria-label="Secondary navigation"
        class="prime-secondary-nav"
      >
        <ul
          class="usa-nav__secondary-links prime-nav"
        >
          <li
            class="usa-nav__secondary-item"
          >
            <a
              class=""
              href="/results/1"
            >
              View test results
            </a>
          </li>
          <li
            class="usa-nav__secondary-item"
          >
            <a
              class=""
              href="/results/upload/submit"
            >
              Upload spreadsheet
            </a>
          </li>
          <li
            class="usa-nav__secondary-item"
          >
            <a
              class=""
              href="/results/upload/submissions"
            >
              View upload history
            </a>
          </li>
        </ul>
      </nav>
      Testing 123
    </div>
  </div>
</div>
`;

exports[`ResultsNav doesnt display the results sub nav when user doesnt have the permission 1`] = `
<div>
  <div
    class="prime-home flex-1"
  >
    <div
      class="grid-container results-wide-container"
    >
      <nav
        aria-label="Secondary navigation"
        class="prime-secondary-nav"
      >
        <ul
          class="usa-nav__secondary-links prime-nav"
        >
          <li
            class="usa-nav__secondary-item"
          >
            <a
              class=""
              href="/results/1"
            >
              View test results
            </a>
          </li>
          <li
            class="usa-nav__secondary-item"
          >
            <a
              class=""
              href="/results/upload/submit"
            >
              Upload spreadsheet
            </a>
          </li>
          <li
            class="usa-nav__secondary-item"
          >
            <a
              class=""
              href="/results/upload/submissions"
            >
              View upload history
            </a>
          </li>
        </ul>
      </nav>
      Testing 123
    </div>
  </div>
</div>
`;
